src-vue/overlays/BootingOverlay.vue

```diff
@@ -84,17 +84,23 @@ localStorage.removeItem('ConfigRestore');
 async function applyRestoredServer(details: string) {
   try {
     const restoreData = JSON.parse(details) as Record<keyof IConfig, string>;
-    if (restoreData.serverDetails) {
-      const parseResult = ConfigServerDetailsSchema.safeParse(JsonExt.parse(restoreData.serverDetails));
-      console.log(parseResult);
-      if (!parseResult.success) return;
+    for (const key of Object.keys(restoreData) as (keyof IConfig)[]) {
+      let value = restoreData[key] as any;
+      if (key === 'miningAccountHadPreviousLife' || key === 'miningAccountPreviousHistory' || key == 'version') {
+        continue;
+      }
```

Comment:
Where are these being set? I only see where `serverDetails` and `hasReadMiningInstructions` is being set.

Reply:
Typescript demanded this because they're read only

Reply:
> Is it because you told TS it is `as (keyof IConfig)` when it's really `as (keyof IConfigRestoreData)[]`?
> 
> ```
> interface IConfigRestoreData { 
>   serverDetails: IConfig['serverDetails'; 
>   hasReadMiningInstructions: IConfig[`hasReadMiningInstructions`];
> }
> ```

Yeah, I assumed we might change the keys that go into restore in future to include other parts of the config, so i left it as all keys